synapse/storage/_base.py

```diff
@@ -69,6 +69,7 @@ def _invalidate_state_caches(
             self._attempt_to_invalidate_cache("is_host_joined", (room_id, host))
 
         self._attempt_to_invalidate_cache("get_users_in_room", (room_id,))
+        self._attempt_to_invalidate_cache("get_users_in_room_with_profiles", (room_id,))
```

Comment:
So, err, we didn't ever invalidate this cache. I think we only ever used this function when handling visibility changes in the user directory, so I think its probably fairly unlikely that this bit anyone...

https://github.com/matrix-org/synapse/blob/c571736c6ca5d1d2d9bf7cd9b717465d446ac7b3/synapse/handlers/user_directory.py#L305

Reply:
I guess the question is whether we want to pull this into a separate PR?

Reply:
"meh". I woudn't bother.

You could always stick in a `.bugfix` as well as a `.feature` newsfile if you're feeling keen.